misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobConsumer.kt

```diff
@@ -64,6 +66,50 @@ internal class SqsJobConsumer @Inject internal constructor(
     }
   }
 
+  /**
+   * Spawn a new thread that will consume the [queueName] SQS attributes and record them as metrics
+   *
+   * The metric recorded are:
+   * - jobs_sqs_aproximate_number_of_messages
+   * - jobs_sqs_aproximate_number_of_messages_not_visible
+   */
+  fun subscribeToMetrics(queueName: QueueName) {
+    log.info {
+      "recording metrics for queue ${queueName.value}"
+    }
+
+    taskQueue.scheduleWithBackoff(Duration.ZERO) {
+      val queue = queues[queueName]
+      val lease = leaseManager.requestLease("sqs-queue-attributes-${queue.queueName}")
+      if (lease.checkHeld()) {
+        val attributes = try {
+          queue.call { client ->
+            val request = GetQueueAttributesRequest()
+                .withQueueUrl(queue.url)
+                .withAttributeNames(
+                    QueueAttributeName.ApproximateNumberOfMessages,
+                    QueueAttributeName.ApproximateNumberOfMessagesNotVisible)
+            val response = client.getQueueAttributes(request)
+            response.attributes
+          }
+        } catch (e: ClientExecutionTimeoutException) {
+          log.info("timed out long polling for queue attributes: ${queue.queueName}")
```

Comment:
you shouldn't be long polling for queue attributes, this should have a reasonable timeout, like 5s and there isn't a reason to catch this exception because it's not normal.

we catch the Exception when receiving messages because we use long polling and sometimes the long poll exceeds the timeout, but we don't care.